mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -643,6 +645,14 @@ LogicalResult getTuningProblemStr(rock::RockGemmWrapperInterface gemmIF,
   KernelType opType = gemmIF.getKernelType();
   Operation *gemmOp = gemmIF.getOperation();
 
+  auto f8TypeStr = [](const Type &type) -> std::optional<StringRef> {
+    if (type.isFloat8E4M3FNUZ() || type.isFloat8E4M3FN())
```

Comment:
Note, you might want this to be `std::optional<StringLiteral>` to make the lifetime situation clearer